django/contrib/admin/static/admin/js/core.js

```diff
@@ -159,7 +159,12 @@ function findPosY(obj) {
                 year = date[i];
                 break;
             case "%y":
-                year = date[i];
+                // 50 is cut off year. 50 to 99 means year 19xx.  00 to 49 means year 20xx.
+                if (parseInt(date[i], 10) >= 50) {
+                    year = date[i];
+                } else {
+                    year = (new Date(Date.UTC(date[i], 0))).getUTCFullYear() + 100;
+                }
```

Comment:
Python's `datetime.strptime()` doesn't work that way. It should be based on RFC-7231, i.e. year that appears to be more than 50 years in the future is interpreted as representing the past. For example:
 - `27/09/68` -> `2068-09-27`
 - `27/09/70` -> `1970-09-27`

Reply:
https://github.com/python/cpython/blob/fdafa1d0ed0a8930b52ee81e57c931cc4d5c2388/Lib/_strptime.py#L376

So if the the %y is [0, 68] then it is 2000 else it is 1900 so the condition is quiet straighforward

Reply:
We can follow the Open Group specification (as CPython) or RFC7231 (see 7b5f8acb9e6395a1660dd7bfeb365866ca8ef47c). I'm not sure which one is better :thinking:

Reply:
IMO should go with Open Group specification (as CPython), because this date is going to be consumed by `datetime.strptime()`, so it makes more sense that parsed date is complaint with it.

Reply:
Or should we patch strptime with the logic so that it is more flexible to the changing centuries?

Reply:
Moving forward with the Open Group specification sounds good to me.